applications/NurbsBrepApplication/custom_python/add_custom_utilities_to_python.cpp

```diff
@@ -20,17 +20,192 @@
+            .def("DerivativesAt", &Type::DerivativesAt,
+                "T"_a,
+                "Order"_a)
+            .def("ValueAt", (double (Type::*)(const Variable<double>&,
+                const double)) &Type::ValueAt<double>,
+                "Variable"_a,
+                "T"_a)
+            .def("ValueAt", (std::vector<double> (Type::*)(
+                const Variable<double>&, const double, const int))
+                &Type::ValueAt<double>,
+                "Variable"_a,
+                "T"_a,
+                "Order"_a)
+            .def("ValueAt", (Vector3D (Type::*)(const Variable<Vector3D>&,
+                const double)) &Type::ValueAt<Vector3D>,
+                "Variable"_a,
+                "T"_a)
+            .def("ValueAt", (std::vector<Vector3D> (Type::*)(
+                const Variable<Vector3D>&, const double, const int))
+                &Type::ValueAt<Vector3D>,
+                "Variable"_a,
+                "T"_a,
+                "Order"_a)
+            .def("ValueAt", (double (Type::*)(const VariableComponent&,
+                const double)) &Type::ValueAt<double, VariableComponent>,
+                "Variable"_a,
+                "T"_a)
+            .def("ValueAt", (std::vector<double> (Type::*)(
+                const VariableComponent&, const double,
+                const int)) &Type::ValueAt<double, VariableComponent>,
+                "Variable"_a,
+                "T"_a,
+                "Order"_a)
+        ;
+    }
+
+    // register NodeSurfaceGeometry
+    {
+        using Type = NodeSurfaceGeometry3D;
+        using Pointer = std::shared_ptr<Type>;
+
+        using VariableComponent = Kratos::VariableComponent<
+            Kratos::VectorComponentAdaptor<Kratos::array_1d<double, 3>>>;
+
+        pybind11::class_<Type, Pointer>(m, "NodeSurfaceGeometry3D")
+            .def(pybind11::init<int, int, int, int>(),
+                "DegreeU"_a,
+                "DegreeV"_a,
+                "NumberOfNodesU"_a,
+                "NumberOfNodesV"_a)
+            .def("SetKnotU", &Type::SetKnotU,
+                "Index"_a,
+                "Value"_a)
+            .def("SetKnotV", &Type::SetKnotV,
+                "Index"_a,
+                "Value"_a)
+            .def("Node", &Type::Node,
+                "IndexU"_a,
+                "IndexV"_a)
+            .def("SetNode", &Type::SetNode,
+                "IndexU"_a,
+                "IndexV"_a,
+                "Value"_a)
+            .def("Pole", &Type::Pole,
+                "IndexU"_a,
+                "IndexV"_a)
+            .def("SetPole", &Type::SetPole,
+                "IndexU"_a,
+                "IndexV"_a,
+                "Value"_a)
+            .def("Weight", &Type::Weight,
+                "IndexU"_a,
+                "IndexV"_a)
+            .def("SetWeight", &Type::SetWeight,
+                "IndexU"_a,
+                "IndexV"_a,
+                "Value"_a)
+            .def("PointAt", &Type::PointAt,
+                "U"_a,
```

Comment:
just curious, what does the _a do?

Reply:
Just a shortcut for `py::arg("named_argument")`. I think named arguments make code much more easier to read.

e.g. `curve.PointAt(T=0.5)`